src/main/java/com/yelp/nrtsearch/server/luceneserver/field/IntFieldDef.java

```diff
@@ -32,10 +33,13 @@
 import org.apache.lucene.search.SortField;
 
 /** Field class for 'INT' field type. */
-public class IntFieldDef extends NumberFieldDef {
+public class IntFieldDef extends NumberFieldDef implements Keyable {
```

Comment:
Numeric fields will only index the default FieldType if store is true. To index the key when store is false, you will need to override parseDocumentField() and add this field if isKey and not store.

Reply:
To keep things simple, I have made `store=true` as mandatory for Keyable fields. We don't really need it in docvalues, but have to guarantee indexing which is taken care of by making `store=true`.